### ISSUE TITLE
Duplicate GEMVfatStatusDigi in cmssw

### ISSUE THREAD

**Issue by davidlange6** [cms-sw/cmssw#30586]
we have two, which differ GEMVfatStatusDigi implementations in CMSSW. One in DataFormats/GEMDigi (unused at this moment) and one in EventFilter/GEMRawToDigi. Perhaps other classes are duplicated between these packages. Could this be cleaned up asap? 

Perhaps related there is a dictionary definition in EventFilter/GEMRawToDigi that would seem to be against usual CMSSW coding practices.

**Comment by davidlange6** [comment]
assign reconstruction

**Comment by perrotta** [comment]
@jshlee @watson-ij

**Comment by jshlee** [comment]
Yes, one of them is old and should be removed.
We will do this soon.

**Comment by davidlange6** [comment]
Thanks - please have things end up in DataFormats as is standard across CMS. 

I worry that this will break reading of any data with these classes saved, so probably good to do it asap.

**Comment by watson-ij** [comment]
#30547 is merged and removes the unused vfatstatusdigi from DataFormats. I will check and then make a PR to also remove DataFormats/GEMDigi/GEMAMCStatusDigi and GEMGEBStatusDigi which also are unused now, and I'm discussing with @jshlee about what objects to then move from EventFilter/ into DataFormats/.

**Comment by slava77** [comment]
> Perhaps related there is a dictionary definition in EventFilter/GEMRawToDigi that would seem to be against usual CMSSW coding practices.

IIUC, the data products defined in EventFilter/GEMRawToDigi/ are not meant for persistency.
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCreatingNewProducts#Restrictions_on_a_package_defini
```
However, for a transient only product, you may choose to locate your package 
outside of one of these [DataFormats*] subsystems.
```
So, it would be OK to define the dictionaries here, more appropriately with a transient flag set.

However, EventFilter/GEMRawToDigi/src/classes_def.xml is the only case in the EventFilter subsystem, so, the relocation is acceptable.

**Comment by perrotta** [comment]
+1
Fixed by #30547, #30609 and #30740
